You will be provided with a partial code base and an issue statement explaining a problem to resolve.

<issue>
UNKNOWN_TABLE does not use the output format when using http_write_exception_in_output_format
**Describe what's wrong**

ThrowIf works fine and the exception is returned as part of the JSON response:
```
curl -i 'localhost:8123/?http_write_exception_in_output_format=1&query=SELECT%20throwIf(1)%20FROM%20system.numbers%20FORMAT%20JSON'
HTTP/1.1 200 OK
Date: Mon, 09 Oct 2023 16:18:48 GMT
Connection: Keep-Alive
Content-Type: application/json; charset=UTF-8
X-ClickHouse-Server-Display-Name: Mordor
Transfer-Encoding: chunked
X-ClickHouse-Query-Id: 18a3e4f2-b041-49e7-b9ba-850018ab1eb5
X-ClickHouse-Format: JSON
X-ClickHouse-Timezone: Europe/Madrid
Keep-Alive: timeout=30
X-ClickHouse-Summary: {"read_rows":"65409","read_bytes":"523272","written_rows":"0","written_bytes":"0","total_rows_to_read":"0","result_rows":"0","result_bytes":"0","elapsed_ns":"1004645"}

{
        "meta":
        [
                {
                        "name": "throwIf(1)",
                        "type": "UInt8"
                }
        ],

        "data":
        [

        ],

        "rows": 0,

        "exception": "Code: 395. DB::Exception: Value passed to 'throwIf' function is non-zero: while executing 'FUNCTION throwIf(1 :: 0) -> throwIf(1) UInt8 : 1'. (FUNCTION_THROW_IF_VALUE_IS_NON_ZERO) (version 23.10.1.1)"
}
```

But when the table doesn't exists you still get a plain text message and not JSON output, which is inconsistent:

```
curl -i 'localhost:8123/?http_write_exception_in_output_format=1&query=SELECT%20throwIf(1)%20FROM%20not_existing%20FORMAT%20JSON'
HTTP/1.1 404 Not Found
Date: Mon, 09 Oct 2023 16:18:39 GMT
Connection: Keep-Alive
Content-Type: text/plain; charset=UTF-8
X-ClickHouse-Server-Display-Name: Mordor
Transfer-Encoding: chunked
X-ClickHouse-Query-Id: b2fe023b-250b-44df-834f-359660364044
X-ClickHouse-Timezone: Europe/Madrid
X-ClickHouse-Exception-Code: 60
Keep-Alive: timeout=30
X-ClickHouse-Summary: {"read_rows":"0","read_bytes":"0","written_rows":"0","written_bytes":"0","total_rows_to_read":"0","result_rows":"0","result_bytes":"0","elapsed_ns":"516343"}

Code: 60. DB::Exception: Table default.not_existing does not exist. (UNKNOWN_TABLE) (version 23.10.1.1)
```

Ideally it should always use the output format.

Related to https://github.com/ClickHouse/ClickHouse/issues/6272

**Does it reproduce on recent release?**

Tested on master/HEAD
</issue>

I need you to solve the provided issue by generating a code fix that can be applied directly to the repository

Respond below:
